tests/server/util.c

```diff
@@ -55,9 +58,14 @@
 #define EINVAL  22 /* errno.h value */
 #endif
 
+/* MinGW with w32api version < 3.6 declared in6addr_any as extern,
+   but lacked the definition */
 #if defined(ENABLE_IPV6) && defined(__MINGW32__)
+#if (__W32API_MAJOR_VERSION < 3) || \
+    ((__W32API_MAJOR_VERSION == 3) && (__W32API_MINOR_VERSION < 6))
 const struct in6_addr in6addr_any = {{ IN6ADDR_ANY_INIT }};
```

Comment:
mingw-w64 [mingw-w64-headers/include/ws2tcpip.h](https://github.com/mirror/mingw-w64/blob/v5.0.2/mingw-w64-headers/include/ws2tcpip.h#L76):
~~~c
#define IN6ADDR_ANY_INIT { { { 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 } } }
~~~
libcurl [tests/server/util.c](https://github.com/curl/curl/blob/curl-7_53_1/tests/server/util.c#L59):
~~~c
const struct in6_addr in6addr_any = {{ IN6ADDR_ANY_INIT }};
~~~
The extra braces appear to have been added to util.c in 14424f7. Original mingw defines that macro as {0} and it has to be {{{0}}} to avoid a missing braces warning at some levels.

Re fix, it's fine but the comment is superfluous since the check makes it obvious.